okhttp-clients/src/test/java/com/palantir/remoting3/okhttp/OkHttpClientsTest.java

```diff
@@ -314,16 +264,59 @@ public void interceptsAndHandlesQos_endToEnd_memorizedCurrentUrlBetweenCalls() t
         server2.enqueue(new MockResponse().setResponseCode(200).setBody("foo"));
 
         Call call = client.newCall(new Request.Builder().url(url).build());
-        assertThat(call.execute().body().string()).isEqualTo("foo");
+        assertThat(execute(call).body().string()).isEqualTo("foo");
 
         assertThat(server.getRequestCount()).isEqualTo(1);
         assertThat(server2.getRequestCount()).isEqualTo(2);
     }
 
     private OkHttpClient createRetryingClient(int maxNumRetries) {
+        return createRetryingClient(maxNumRetries, Duration.ofMillis(10));
+    }
+
+    private OkHttpClient createRetryingClient(int maxNumRetries, Duration backoffSlotSize) {
         return OkHttpClients.withStableUris(
-                ClientConfiguration.builder().from(createTestConfig(url)).maxNumRetries(maxNumRetries).build(),
+                ClientConfiguration.builder().from(createTestConfig(url))
+                        .maxNumRetries(maxNumRetries)
+                        .backoffSlotSize(backoffSlotSize)
+                        .build(),
                 AGENT,
                 OkHttpClientsTest.class);
     }
+
+    private Response execute(Call call) throws Exception {
+        return callExecutor.execute(call);
+    }
+
+    interface CallExecutor {
+
+        CallExecutor BLOCKING = call -> call.execute();
+
+        CallExecutor ASYNC = call -> {
+            FutureCallback callback = new FutureCallback();
+            call.enqueue(callback);
+            try {
+                return callback.get();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            } catch (ExecutionException e) {
+                throw (Exception) e.getCause();
```

Comment:
What if cause is an Error?

Reply:
Or general Throwable?

Reply:
This is a hack because checkstyle doesn't let you declare `throws Throwable` =/

Reply:
But also, the callback interface only accepts `IOException`, so this will always be safe